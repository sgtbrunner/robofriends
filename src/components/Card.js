import React from 'react';

const Card = ({ name, email }) => {
 	return (
   <div className="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green">
     <img alt="robot" src={`https://robohash.org/${name}`} />
     <h2>{name}</h2>
     <p>{email}</p>
   </div>
 	);
};

export default Card;
