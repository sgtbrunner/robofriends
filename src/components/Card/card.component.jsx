import React from 'react';

const Card = ({ name, email, id }) => (
  <div className="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green">
    <img alt="robot" src={`https://robohash.org/${name + email + id}`} />
    <h2>{name}</h2>
    <p>{email}</p>
  </div>
);

export default Card;
