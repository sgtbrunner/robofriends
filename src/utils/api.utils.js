export const USERS_URL = 'https://jsonplaceholder.typicode.com/users';

export default {
  getUsers: () =>
    fetch(USERS_URL).then((response) => {
      if (response.ok) return response.json();
      throw new Error('Something went wrong');
    }),
};
