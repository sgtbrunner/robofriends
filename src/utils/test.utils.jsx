import React from 'react';
import { render } from '@testing-library/react';

export const mountComponent = ({ component, props = {} }) => {
  const Component = component;
  return render(<Component {...props} />);
};

export default null;
