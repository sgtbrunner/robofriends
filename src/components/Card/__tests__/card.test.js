import { screen } from '@testing-library/react';

import { mountComponent } from '../../../utils/test.utils';
import Card from '../index';

const props = {
  id: 1,
  name: 'Baby Yoda',
  email: 'baby.yoda@newrepublic.com',
};

describe('Card component', () => {
  describe('render', () => {
    test('expect Card component to match snapshot', () => {
      const { container } = mountComponent({ component: Card, props });
      expect(container).toMatchSnapshot();
    });

    test('check component child elements', () => {
      mountComponent({ component: Card, props });

      // check image
      const image = screen.getByRole('img', { name: 'robot' });
      expect(image).toBeInTheDocument();

      // check name
      const name = screen.getByRole('heading', { name: props.name });
      expect(name).toBeInTheDocument();

      // check email
      const email = screen.getByText(props.email);
      expect(email).toBeInTheDocument();
    });
  });
});
