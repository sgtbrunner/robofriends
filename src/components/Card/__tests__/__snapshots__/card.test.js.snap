// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component render expect Card component to match snapshot 1`] = `
<div>
  <div
    class="dib br3 pa1 ma2 grow bw2 shadow-5 bg-light-green"
  >
    <img
      alt="robot"
      src="https://robohash.org/Baby Yodababy.yoda@newrepublic.com1"
    />
    <h2>
      Baby Yoda
    </h2>
    <p>
      baby.yoda@newrepublic.com
    </p>
  </div>
</div>
`;
